move/interchain_token_service/sources/types/coin_info.move

```diff
@@ -21,13 +22,27 @@ module interchain_token_service::coin_info {
         }
     }
 
-    /// Create a new coin info from the given `CoinMetadata` object.
+    /// Create a new coin info from the given `CoinMetadata` object and publicly freeze the metadata object.
     public fun from_metadata<T>(metadata: CoinMetadata<T>): CoinInfo<T> {
-        CoinInfo {
+        let coin_info = CoinInfo {
             name: metadata.get_name(),
             symbol: metadata.get_symbol(),
             decimals: metadata.get_decimals(),
-            metadata: option::some(metadata),
+            metadata: option::none(),
+        };
+        transfer::public_freeze_object(metadata);
```

Comment:
call release_metadata in the caller instead of doing this in a constructor

Reply:
I think adding this would have unintended behaviour as `CoinMetadata` is owned by the function at this point in the call stack and needs to be used somehow or destroyed by the function, because it's og purpose was to store the value. Since we can't change the signature or return type, and only <T> has drop, the `CoinMetadata` would get lost.

Reply:
Yeah, you're right, makes sense to freeze it here since we can't remove this method